apollo/handlers/user.py

```diff
@@ -5,18 +5,22 @@
 from sqlalchemy.orm import Session
 
 from apollo.lib.exceptions import HTTPException
-from apollo.lib.hash import hash_plaintext
+from apollo.lib.hash import hash_plaintext, compare_plaintext_to_hash
 from apollo.lib.router import SecureRouter
-from apollo.lib.schemas.user import CreateUserSchema, UserSchema
-from apollo.lib.security import Allow, Admin, Human
+from apollo.lib.schemas.user import (
+    CreateUserSchema, UserSchema, UpdateUserSchema)
+from apollo.lib.security import (Allow, Admin, Human, Authenticated,
+                                 Uninitialized)
 from apollo.models import get_session, save, delete
 from apollo.models.user import User, get_user_by_id, list_users as query_users
 
 router = SecureRouter([
     (Allow, Admin, 'user.post'),
     (Allow, Admin, 'user.delete'),
     (Allow, Admin, 'user.list'),
-    (Allow, Human, 'user.get_current')
+    (Allow, Human, 'user.get_current'),
+    (Allow, Authenticated, 'user.patch'),
```

Comment:
```suggestion
    (Allow, Authenticated, 'user.patch_self'),
```
Or whatever, to avoid confusion.